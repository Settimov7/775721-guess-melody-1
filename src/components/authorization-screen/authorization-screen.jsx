import * as React from 'react';
import {connect} from "react-redux";
import PropTypes from "prop-types";

import {Operation} from "../../reducer/user/user";

export class AuthorizationScreen extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      name: ``,
      password: ``,
    };

    this._inputChangeHandle = this._inputChangeHandle.bind(this);
    this._submitHandle = this._submitHandle.bind(this);
  }

  render() {
    const {name, password} = this.state;

    return (
      <section className="login">
        <div className="login__logo">
          <img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83" />
        </div>
        <h2 className="login__title">Необходима авторизация</h2>
        <p className="login__text">Представтесь!</p>
        <form
          className="login__form"
          onSubmit={this._submitHandle}
        >
          <p className="login__field">
            <label className="login__label" htmlFor="name">Логин</label>
            <input
              className="login__input"
              type="text"
              name="name"
              id="name"
              required
              value={name}
              onChange={this._inputChangeHandle}
            />
          </p>
          <p className="login__field">
            <label className="login__label" htmlFor="password">Пароль</label>
            <input
              className="login__input"
              type="text"
              name="password"
              id="password"
              required
              value={password}
              onChange={this._inputChangeHandle}
            />
            <span className="login__error">Неверный пароль</span>
          </p>
          <button className="login__button button" type="submit">Войти</button>
        </form>
      </section>
    );
  }

  _inputChangeHandle(evt) {
    const {value, name} = evt.target;

    this.setState({
      [name]: value,
    });
  }

  _submitHandle(evt) {
    evt.preventDefault();

    const {name, password} = this.state;
    const {login} = this.props;

    login(name, password);
  }
}

AuthorizationScreen.propTypes = {
  login: PropTypes.func.isRequired,
};

const mapDispatchToProps = {
  login: Operation.login,
};

export default connect(null, mapDispatchToProps)(AuthorizationScreen);
