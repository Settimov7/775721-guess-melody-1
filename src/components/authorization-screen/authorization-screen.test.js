import * as React from 'react';
import renderer from 'react-test-renderer';

import {AuthorizationScreen} from "./authorization-screen";

it(`Authorization screen correctly renders`, () => {
  const login = jest.fn();
  const props = {
    login,
  };
  const authorizationScreen = renderer.create(<AuthorizationScreen {...props} />).toJSON();

  expect(authorizationScreen).toMatchSnapshot();
});
