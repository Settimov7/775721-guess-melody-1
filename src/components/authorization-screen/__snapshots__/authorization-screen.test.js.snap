// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization screen correctly renders 1`] = `
<section
  className="login"
>
  <div
    className="login__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="login__title"
  >
    Необходима авторизация
  </h2>
  <p
    className="login__text"
  >
    Представтесь!
  </p>
  <form
    className="login__form"
    onSubmit={[Function]}
  >
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="name"
      >
        Логин
      </label>
      <input
        className="login__input"
        id="name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </p>
    <p
      className="login__field"
    >
      <label
        className="login__label"
        htmlFor="password"
      >
        Пароль
      </label>
      <input
        className="login__input"
        id="password"
        name="password"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <span
        className="login__error"
      >
        Неверный пароль
      </span>
    </p>
    <button
      className="login__button button"
      type="submit"
    >
      Войти
    </button>
  </form>
</section>
`;
